modules/messages/src/outbound_lane.rs

```diff
@@ -150,41 +145,17 @@ impl<S: OutboundLaneStorage> OutboundLane<S> {
 
 		ensure_unrewarded_relayers_are_correct(confirmed_messages.end, relayers)?;
 
+		// prune all confirmed messages
```

Comment:
Two things to note here:
1) this removal is accounted by our benchmarks (see changes in weights db writes);
2) we do weight refund in delivery confirmation transaction, but it doesn't need to be changed because it does refund based on `total_messages` from the call arguments and `confirmed_messages` range, returned by this function. So all should be fine here